spec/importc.dd

```diff
@@ -245,6 +247,77 @@ $(H2 $(LNAME2 extensions, Extensions))
     $(P Arrays can have `register` storage class, and may be enregistered by the compiler. C11 6.3.2.1-3)
 
 
+    $(H3 $(LNAME2 __import, Import Declarations))
+
+    $(P Modules can be imported with a $(I CImportDeclaration):)
+
+$(GRAMMAR
+$(GNAME CImportDeclaration):
+    $(D __import) $(GLINK2 module, ImportList) $(D ;)
+)
+
+    $(P Imports enable ImportC code to directly access D declarations and functions
+    without the necessity of creating a $(TT .h) file representing those declarations.
+    The tedium and brittleness of keeping the $(TT .h) file up-to-date with the D
+    declarations is eliminated.
+    D functions are available to be inlined.
+    )
+
+    $(P Imports also enable ImportC code to directly import other C files without
+    needing to create a .h file for them, either.
+    Imported C functions become available to be inlined.
+    )
+
+    $(P The $(I ImportList) works the same as it does for D.)
+
+    $(P The ordering of $(I CImportDeclaration)s has no significance.)
+
+    $(P An ImportC file can be imported, the name of the C file to be
+    imported is derived from the module name.)
+
+    $(P All the global symbols in the ImportC file become available to the
+    importing module.)
+
+    $(P If a name is referred to in the importing file is not found,
+    the global symbols in each imported file are searched for the name.
+    If it is found in exactly one module, that becomes the resolution of the
+    name. If it is found in multiple modules, it is an error.)
+
+    $(NOTE Since ImportC has no scope resolution operator, only global symbols
+    can be found, and a qualification cannot be added to specifiy which module
+    a symbols is in.)
+
+    $(P Preprocessor symbols in the imported module are not available to the
+    importing module, and preprocessing symbols in the importing file are not
+    available to the imported module.)
+
+    $(P A D module can be imported, in the same manner as that
+    of a $(GLINK2 module, ImportDeclaration).)
+
+    $(P Imports can be circular.)
+
+$(CCODE
+__import core.stdc.stdarg; // get D declaration of va_list
```

Comment:
I think this example should be something like hello world instead since `va_list`s can't actually be manipulated (with all capabilities required by the standard) without a shim.

Reply:
The rest is good.

Reply:
I'm not understanding what you mean. User shims aren't required to do stdarg stuff, that is taken care of by the default importation of __builtins.c.

Reply:
`va_arg` is a macro, that presumably won't work for a while?

Reply:
va_arg is a template.

Reply:
`va_arg` as in https://en.cppreference.com/w/c/variadic/va_arg is a macro.

I will be opening a PR to add basic preprocessor support soon